plugins/pixart-tp/fu-pxi-tp.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2025 Harris Tai <harris_tai@pixart.com>
+// Copyright 2025 Micky Hsieh <micky_hsieh@pixart.com>
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+// TF write-simple command header:
+//   REPORT_ID__PASS_THROUGH
+//   TF_FRAME_PREAMBLE
+//   SLAVE_ADDRESS
+//   TF_FUNC_WRITE_SIMPLE
+//   addr (LE16)
+//   len  (LE16)
+#[derive(New)]
+#[repr(C, packed)]
+struct FuStructPxiTfWriteSimpleCmd {
+    report_id: u8,
+    preamble: u8,
+    slave_addr: u8,
+    func: u8,
+    addr: u16le,
+    len: u16le,
+}
+
+// TF write-with-packet command header:
+//   REPORT_ID__PASS_THROUGH
+//   TF_FRAME_PREAMBLE
+//   SLAVE_ADDRESS
+//   TF_FUNC_WRITE_WITH_PACK
+//   addr         (LE16)
+//   datalen      (LE16)  // payload bytes: 2 bytes total + 2 bytes index + data
+//   packet_total (LE16)
+//   packet_index (LE16)
+#[derive(New)]
+#[repr(C, packed)]
+struct FuStructPxiTfWritePacketCmd {
+    report_id: u8,
```

Comment:
if this is always REPORT_ID__PASS_THROUGH then you can define it here -- add `Defaults` to the list in `#[derive()]` and then use `report_id: u8 == 0xCC` -- this also means the `fu_struct_pxi_tf_write_packet_cmd_set_report_id()` command won't be generated anymore, because `fu_struct_pxi_tf_write_packet_cmd_new()` will set the value by default.